src/pineko/check.py

```diff
@@ -118,36 +123,50 @@ def contains_fact(grid, max_as, max_al):
         is fact scale-variation available for as
     bool
         is fact scale-variation available for al
+    bool
+        is the central as order reached by the grid
+    bool
+        is the central al order reached by the grid
     """
     order_array = np.array([order.as_tuple() for order in grid.orders()])
     order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
     order_list = order_array[order_mask]
     as_orders = []
     al_orders = []
+    min_al = min([ord[1] for ord in order_list])
     for order in order_list:
-        as_orders.append(order[0])
-        al_orders.append(order[1])
-    min_as = min(as_orders)
-    min_al = min(al_orders)
+        if order[1] == min_al:
+            as_orders.append(order)
+        if order[1] != 0:
+            al_orders.append(order)
+    min_as = min([ord[0] for ord in as_orders]) if len(as_orders) != 0 else 0
+    min_al = min([ord[0] for ord in al_orders]) if len(al_orders) != 0 else 0
     order_as_is_present = False
     order_al_is_present = False
     sv_as_present = False
     sv_al_present = False
-    for order in order_list:
-        # fact sv starts at NLO with respect to the first non zero order
-        if order[0] == min_as + 1:
+    # fact sv starts at NLO with respect to the first non zero order
+    add_order_as = 1 + (max_as - 2)
+    add_order_al = 1 + (max_al - 2)
+    if max_as == 1:
+        add_order_as = 1
+    if max_al == 1:
+        add_order_al = 1
+    for order in as_orders:
+        if order[0] == min_as + add_order_as:
             order_as_is_present = True
             if order[-1] != 0:
                 sv_as_present = True
-        if order[1] == min_al + 1:
+    for order in al_orders:
+        if order[1] == min_al + add_order_al:
             order_al_is_present = True
             if order[-1] != 0:
                 sv_al_present = True
     if not order_as_is_present:
         sv_as_present = True
     if not order_al_is_present:
         sv_al_present = True
-    return sv_as_present, sv_al_present
+    return sv_as_present, sv_al_present, order_as_is_present, order_al_is_present
```

Comment:
Can you split this function in 4 different functions? Or at least 2, since `as` and `al` are fully decoupled (in particular, they would be the same function, with a switch for one or the other)

They should take as input an `order_list`, and the relevant `max` (call it `max_a`, if you wish).

Reply:
The problem is that `as` and `al` are not decoupled. In particular the function that is coupling them is `order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)` and it makes sense because, depending on `max_al`, you need to take into account different orders. Now, it is true that if there is no ren_sv taking `max_al=0`, there won't be for `max_al != 0` as well, but I believe we should not rely on this. This is the reason why they are coupled.

Reply:
Of course the same applies for `contains_fact`